### PR TITLE
feat(gatsby-plugin-sass): Add ability to override sassRuleTest and sassRuleModulesTest

### PR DESCRIPTION
**Pull request by tkbky**
##  Description

This PR adds 2 new options to gatsby-plugin-sass
- `sassRuleTest` to configure regex for sass files.
- `sassRuleModulesTest` to configure regex for css modules files.

## Related Issues

Fixes [#15879](https://github.com/gatsbyjs/gatsby/issues/15879)

### CODE REVIEW

File: packages/gatsby-plugin-sass/README.md [hunk 1]
```diff
@@ -126,6 +126,25 @@ plugins: [
...
+```javascript
+// in gatsby-config.js
+plugins: [
+  {
+    resolve: `gatsby-plugin-sass`,
+    options: {
+      // Override the file regex for SASS
+      sassRuleTest: /\.global\.s(a|c)ss$/,
+      // Override the file regex for CSS modules
+      sassRuleModulesTest: /\.global\.s(a|c)ss$/,
```
Comment (abumalick) [inline packages/gatsby-plugin-sass/README.md]: I would change from global to `mod` here, it is clearer

File: packages/gatsby-plugin-sass/src/__tests__/gatsby-node.js [hunk 2]
```diff
@@ -34,6 +34,12 @@ describe(`gatsby-plugin-sass`, () => {
           camelCase: false,
         },
       },
+      "sass rule test options": {
+        sassRuleTest: `/\.global\.s(a|c)ss$/`,
+      },
+      "sass rule modules test options": {
+        sassRuleModulesTest: /\.global\.s(a|c)ss$/,
```
Comment (abumalick) [inline packages/gatsby-plugin-sass/src/__tests__/gatsby-node.js]: here too, I would replace the `global` by `mod`

**Comment by sidharthachatterjee** [comment]
Published in `gatsby-plugin-sass@2.1.6`

**Comment by tkbky** [comment]
Thanks @sidharthachatterjee

**Pull request merged**